pennylane/templates/layers/particle_conserving_u1.py

```diff
@@ -0,0 +1,251 @@
+    `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_.
+
+    This template prepares :math:`N`-qubit trial states by applying :math:`D` layers of the
+    entangler block :math:`U_\mathrm{ent}(\vec{\phi}, \vec{\theta})` to the Hartree-Fock
+    state
+
+    .. math::
+
+        \vert \Psi(\vec{\phi}, \vec{\theta}) \rangle = \hat{U}^{(D)}_\mathrm{ent}(\vec{\phi}_D,
+        \vec{\theta}_D) \dots \hat{U}^{(2)}_\mathrm{ent}(\vec{\phi}_2, \vec{\theta}_2)
+        \hat{U}^{(1)}_\mathrm{ent}(\vec{\phi}_1, \vec{\theta}_1) \vert \mathrm{HF}\rangle.
+
+    The circuit implementing the entangler blocks is shown in the figure below:
+
+    |
+
+    .. figure:: ../../_static/templates/layers/particle_conserving_u1.png
+        :align: center
+        :width: 50%
+        :target: javascript:void(0);
+
+    |
+
+    The repeated units across several qubits are shown in dotted boxes. Each layer
+    contains :math:`N-1` particle-conserving two-parameter exchange gates
+    :math:`U_{1,\mathrm{ex}}(\phi, \theta)` that act on pairs of nearest neighbors qubits.
+    The unitary matrix representing :math:`U_{1,\mathrm{ex}}(\phi, \theta)`
+    is given by (see `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_),
+
+    .. math::
+
+        U_{1, \mathrm{ex}}(\phi, \theta) = \left(\begin{array}{cccc}
+        1 & 0 & 0 & 0 \\
+        0 & \mathrm{cos}(\theta) & e^{i\phi} \mathrm{sin}(\theta) & 0 \\
+        0 & e^{-i\phi} \mathrm{sin}(\theta) & -\mathrm{cos}(\theta) & 0 \\
+        0 & 0 & 0 & 1 \\
+        \end{array}\right).
+
+    The figure below shows the circuit decomposing :math:`U_{1, \mathrm{ex}}` in
+    elementary gates. The Pauli matrix :math:`\sigma_z` and single-qubit rotation :math:`R(0, 2 \theta, 0)`
+    apply the Pauli Z operator and an arbitrary rotation on the qubit ``n``,
+
+    |
+
+    .. figure:: ../../_static/templates/layers/u1_decomposition.png
+        :align: center
+        :width: 80%
+        :target: javascript:void(0);
+
+    |
+
+    :math:`U_A(\phi)` is the unitary matrix
+
+    .. math::
+
+        U_A(\phi) = \left(\begin{array}{cc} 0 & e^{-i\phi} \\ e^{-i\phi} & 0 \\ \end{array}\right),
+
+    acting controlledly on the state of qubit ``m`` which is further decomposed in terms of the
+    `quantum operations <https://pennylane.readthedocs.io/en/stable/introduction/operations.html>`_
+    supported by Pennylane,
+
+    |
+
+    .. figure:: ../../_static/templates/layers/ua_decomposition.png
+        :align: center
+        :width: 70%
+        :target: javascript:void(0);
+
+    |
+
+    where,
+
+    |
+
+    .. figure:: ../../_static/templates/layers/phaseshift_decomposition.png
+        :align: center
+        :width: 65%
+        :target: javascript:void(0);
+
+    |
+
+    The quantum circuits above decomposing the unitaries :math:`U_{1,\mathrm{ex}}(\phi, \theta)`
+    and :math:`U_A(\phi)` are implemented by the :func:`~.u1_ex_gate` and :func:`~.decompose_ua`
```

Comment:
Small note: the function references are not created here (I think it might be because these functions are not imported in an `__init__` file of a subpackage). Just a thing to note, might not quite sure about its significance.

Reply:
Yes, that's right. But if we add it this helper function will also show up  in the template's page?. Probably we don't want to do that, right?. Feel free to let me know.

Reply:
Yes, I agree. I think it's fine to either leave it as is or simply just to enclose them with double back ticks (just to indicate that referencing is not attempted here).